src/LibraryViewExtension/LibraryViewCustomization.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Dynamo.Wpf.Interfaces;
+
+namespace Dynamo.LibraryUI
+{
+    class LibraryViewCustomization : ILibraryViewCustomization
+    {
+        public const string DefaultSectionName = "default";
+        private LayoutSpecification root = new LayoutSpecification();
+
+        private void RaiseSepcificationUpdated()
+        {
+            var handler = SpecificationUpdated;
+            if (handler != null) handler(this, EventArgs.Empty);
+        }
+
+        /// <summary>
+        /// Gets a copy of the current library layout specification
+        /// </summary>
+        public LayoutSpecification Specification
+        {
+            get
+            {
+                return root.Clone();
```

Comment:
correct, I also didn't feel right about it. Hence the interface doesn't have Specification property rather it has a method GetSpecication(). The reason we don't want to pass the original specification is that a client can then modify the content of the specification with informing the customization service and we won't be able to refresh the library view. I have addressed this issue by removing this property and providing an explicit setter method.